diff --git a/src/gs-manager.c b/src/gs-manager.c
index 32c2305..f3606b3 100644
--- a/src/gs-manager.c
+++ b/src/gs-manager.c
@@ -555,7 +555,7 @@ background_settings_change_event_cb (GSettings *settings,
                                      gint       n_keys,
                                      GSManager   *manager)
 {
-#if 0
+#if 1
         /* FIXME: since we bind user settings instead of system ones,
          *        watching for changes is no longer valid.
          */
@@ -578,15 +578,15 @@ get_system_settings (void)
          *        Hence, reset to system default values.
          */
         /* TODO: Ideally we would like to bind some other key, screensaver-specific. */
-        settings = g_settings_new ("org.gnome.desktop.background");
+        settings = g_settings_new ("org.gnome.desktop.screensaver");
 
-        g_settings_delay (settings);
+//        g_settings_delay (settings);
 
-        keys = g_settings_list_keys (settings);
-        for (k = keys; *k; k++) {
-                g_settings_reset (settings, *k);
-        }
-        g_strfreev (keys);
+//        keys = g_settings_list_keys (settings);
+//        for (k = keys; *k; k++) {
+//                g_settings_reset (settings, *k);
+//        }
+//        g_strfreev (keys);
 
         return settings;
 }
