diff -aur gnome-screensaver-2.14.3.vanilla/src/gnome-screensaver-command.c gnome-screensaver-2.14.3/src/gnome-screensaver-command.c
--- gnome-screensaver-2.14.3.vanilla/src/gnome-screensaver-command.c	2006-09-13 10:35:41.000000000 +0200
+++ gnome-screensaver-2.14.3/src/gnome-screensaver-command.c	2006-09-13 10:37:46.000000000 +0200
@@ -320,7 +320,7 @@
         loop = g_main_loop_new (NULL, FALSE);
         g_main_loop_run (loop);
 
-        dbus_connection_disconnect (connection);
+        dbus_connection_close (connection);
 
 	return 0;
 }
