diff -Naur gnome-screensaver-0.0.24.orig/savers/gs-theme-engine.c gnome-screensaver-0.0.24/savers/gs-theme-engine.c
--- gnome-screensaver-0.0.24.orig/savers/gs-theme-engine.c	2006-02-06 21:36:59.000000000 +0100
+++ gnome-screensaver-0.0.24/savers/gs-theme-engine.c	2006-02-06 21:38:12.000000000 +0100
@@ -89,7 +89,7 @@
 
         state = (GtkStateType) 0;
         while (state < (GtkStateType) G_N_ELEMENTS (widget->style->bg)) {
-                gtk_widget_modify_bg (widget, state, &color);
+//                gtk_widget_modify_bg (widget, state, &color);
                 state++;
         }
 
