diff --git a/src/gs-manager.c b/src/gs-manager.c
index 32c2305..16546f6 100644
--- a/src/gs-manager.c
+++ b/src/gs-manager.c
@@ -555,13 +555,11 @@ background_settings_change_event_cb (GSettings *settings,
                                      gint       n_keys,
                                      GSManager   *manager)
 {
-#if 0
         /* FIXME: since we bind user settings instead of system ones,
          *        watching for changes is no longer valid.
          */
         gnome_bg_load_from_preferences (manager->priv->bg,
                                         manager->priv->settings);
-#endif
 
         return FALSE;
 }
@@ -570,23 +568,23 @@ static GSettings *
 get_system_settings (void)
 {
         GSettings *settings;
-        gchar **keys;
-        gchar **k;
+//        gchar **keys;
+//        gchar **k;
 
         /* FIXME: we need to bind system settings instead of user but
          *        that's currently impossible, not implemented yet.
          *        Hence, reset to system default values.
          */
         /* TODO: Ideally we would like to bind some other key, screensaver-specific. */
-        settings = g_settings_new ("org.gnome.desktop.background");
+        settings = g_settings_new ("org.gnome.desktop.screensaver");
 
-        g_settings_delay (settings);
+//        g_settings_delay (settings);
 
-        keys = g_settings_list_keys (settings);
-        for (k = keys; *k; k++) {
-                g_settings_reset (settings, *k);
-        }
-        g_strfreev (keys);
+//        keys = g_settings_list_keys (settings);
+//        for (k = keys; *k; k++) {
+//                g_settings_reset (settings, *k);
+//        }
+//        g_strfreev (keys);
 
         return settings;
 }
